tests/core/util/test_jsonify.py

```diff
@@ -15,21 +15,42 @@ class PrimitivesTest(Streamable):
         c: str
         d: bytes
         e: bytes32
+        f: bool
 
-    t = PrimitivesTest(
+    t1 = PrimitivesTest(
         uint32(123),
         None,
         "foobar",
         b"\0\1\0\1",
         bytes32(range(32)),
+        True,
     )
 
-    assert t.to_json_dict() == {
+    assert t1.to_json_dict() == {
         "a": 123,
         "b": None,
         "c": "foobar",
         "d": "0x00010001",
         "e": "0x000102030405060708090a0b0c0d0e0f101112131415161718191a1b1c1d1e1f",
+        "f": True,
```

Comment:
This doesn't test what you want it to.  It also passes with `1` here.  `bool` actually inherits from `int`, for better or `*cough*` worse.  `True` has a value of 1 and `False` has a value of 0.

The following snippets might be a reasonable choice to address this.  I like this form of creating the dictionaries so that pytest's failure output can present the most info.

```python3
def dict_with_types(d):
    return {k: (v, type(v)) for k, v in d.items()}
```

And then assert such as this.

```python3
assert dict_with_types(t1.to_json_dict()) == dict_with_types(e1)
```